spring-cloud-function-adapters/spring-cloud-function-adapter-google-cloud-function/src/main/java/org/springframework/cloud/function/adapter/gcloud/GcloudSpringBootHttpRequestHandler.java

```diff
@@ -0,0 +1,196 @@
+import com.google.gson.Gson;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+
+import org.springframework.cloud.function.context.AbstractSpringFunctionAdapterInitializer;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.support.GenericMessage;
+
+/**
+ * @author Dmitry Solomakha
+ */
+public class GcloudSpringBootHttpRequestHandler<O>
+	extends AbstractSpringFunctionAdapterInitializer<Context> implements HttpFunction {
+
+	Gson gson = new Gson();
+
+	public GcloudSpringBootHttpRequestHandler() {
+		super();
+	}
+
+	public GcloudSpringBootHttpRequestHandler(Class<?> configurationClass) {
+		super(configurationClass);
+	}
+
+	protected Object convert(HttpRequest event) throws IOException {
+		BufferedReader br = event.getReader();
+		StringBuilder sb = new StringBuilder();
+		String line;
+		while ((line = br.readLine()) != null) {
+			sb.append(line).append("\n");
+		}
+
+		String requestBody = sb.toString();
+		if (functionAcceptsMessage()) {
+			return new GenericMessage<>(toOtionalIfEmpty(requestBody), getHeaders(event, requestBody));
+		}
+		else {
+			return toOtionalIfEmpty(requestBody);
+		}
+
+	}
+
+	private Object toOtionalIfEmpty(String requestBody) {
+		return requestBody.isEmpty() ? Optional.empty() : requestBody;
+	}
+
+	private MessageHeaders getHeaders(HttpRequest event, String requestBody) {
+		Map<String, Object> headers = new HashMap<String, Object>();
+
+		if (event.getHeaders() != null) {
+			headers.putAll(event.getHeaders());
+		}
+		if (event.getQueryParameters() != null) {
+			headers.putAll(event.getQueryParameters());
+		}
+		if (event.getUri() != null) {
+			headers.put("path", event.getPath());
+		}
+
+		if (event.getMethod() != null) {
+			headers.put("httpMethod", event.getMethod());
+		}
+
+		headers.put("request", requestBody);
+		return new MessageHeaders(headers);
+	}
+
+	protected boolean functionAcceptsMessage() {
+		return this.getInspector().isMessage(function());
+	}
+
+	@SuppressWarnings("unchecked")
+	protected <T> T result(Object input, Publisher<?> output, HttpResponse resp) {
+		List<T> result = new ArrayList<>();
+		for (Object value : Flux.from(output).toIterable()) {
+			result.add((T) convertOutput1(value, resp));
+		}
+		if (isSingleValue(input) && result.size() == 1) {
+			return result.get(0);
+		}
+		return (T) result;
+	}
+
+
+	private boolean isSingleValue(Object input) {
+		return !(input instanceof Collection);
+	}
+
+	private Flux<?> extract(Object input) {
+		if (input instanceof Collection) {
+			return Flux.fromIterable((Iterable<?>) input);
+		}
+		return Flux.just(input);
+	}
+
+	public void service(HttpRequest httpRequest, HttpResponse httpResponse) throws Exception {
+		Thread.currentThread()
+			.setContextClassLoader(GcloudSpringBootHttpRequestHandler.class.getClassLoader());
+		initialize(new TestExecutionContext());
```

Comment:
Can you just pass in `null` to initialize? If so it would allow you to remove the testExecutionContext.